.github/workflows/nightly.yml

```diff
@@ -0,0 +1,155 @@
+
+          git tag -a $TAG -m "Streamlit nightly $TAG"
+          git push origin $TAG
+      # TODO: Using testing webhook, switch to actual st-open-source channel webhook later
+      - if: ${{ failure() }}
+        name: Nightly Tag Failure Slack Message
+        env: 
+          SLACK_WEBHOOK: ${{ secrets.TEST_SLACK_WEBHOOK }}
+        run: |
+          curl -X POST -H 'Content-type: application/json' --data '{"text":":blobonfire: Nightly job failed to create a tag"}' "$SLACK_WEBHOOK"
+
+  run-python-tests:
+    needs: create-nightly-tag
+    uses: ./.github/workflows/python-versions.yml
+
+  run-javascript-tests:
+    needs: create-nightly-tag
+    uses: ./.github/workflows/js-tests.yml
+
+  run-py-prod-deps-smoke-test:
+    needs: create-nightly-tag
+    uses: ./.github/workflows/py-prod-deps-smoke-test.yml
+
+  run-cypress-tests:
+    needs: create-nightly-tag
+    uses: ./.github/workflows/cypress.yml
+
+  test-status-notification:
+    runs-on: ubuntu-latest
+
+    if: ${{ always() }}
+    needs: [run-python-tests, run-javascript-tests, run-py-prod-deps-smoke-test, run-cypress-tests]
+
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with: 
+          persist-credentials: false
+          submodules: 'recursive'
+      - name: Send Slack notifications for failing tests
+        uses: ./.github/actions/slack
+        env:
+          SLACK_WEBHOOK: ${{ secrets.TEST_SLACK_WEBHOOK }}
+        with:
+          python_result: ${{ needs.run-python-tests.result }}
+          javascript_result: ${{ needs.run-javascript-tests.result }}
+          py_prod_deps_result: ${{ needs.run-py-prod-deps-smoke-test.result }}
+          cypress_result: ${{ needs.run-cypress-tests.result }}
+  
+  create-nightly-build:
+    runs-on: ubuntu-latest
+
+    needs: [run-python-tests, run-javascript-tests, run-py-prod-deps-smoke-test, run-cypress-tests]
+
+    defaults:
+      run:
+        shell: bash -ileo pipefail {0}
+
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with: 
+          persist-credentials: false
+          submodules: 'recursive'
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Setup virtual env
+        uses: ./.github/actions/make_init
+      - name: Verify git tag vs. version
+        run: |
+          cd lib
+          python setup.py verify
+      # TODO: Added the if statement to gate for now
+      - if: ${{ 'true' == 'false' }}
+        name: Initialize .pypirc
+        env:
+          NIGHTLY_PYPI_USERNAME: ${{ secrets.NIGHTLY_PYPI_USERNAME }}
+          NIGHTLY_PYPI_API_TOKEN: ${{ secrets.NIGHTLY_PYPI_API_TOKEN }}
+        run: |
+          cd lib
+          echo -e "[pypi]" >> ~/.pypirc
+          echo -e "username = "$NIGHTLY_PYPI_USERNAME"" >> ~/.pypirc
+          echo -e "password = "$NIGHTLY_PYPI_API_TOKEN"" >> ~/.pypirc
```

Comment:
hhmm.. do we need to do this?

Reply:
+1

I think we'd prefer to do something like setting `TWINE_USERNAME` and `TWINE_PASSWORD` environment variables, which should be equivalent